statuspage_connector.py

```diff
@@ -0,0 +1,569 @@
+
+    def _handle_list_incidents(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        api_key = "API Key {}".format(self._api_key)  # pragma: allowlist secret
+        headers = {'Authorization': api_key}
+
+        parameters = dict()
+        query = param.get('query')
+        limit = param.get('limit')
+        page = param.get('page_offset')
+
+        if query:
+            parameters['q'] = query
+        if limit:
+            parameters['limit'] = limit
+        if page:
+            parameters['page'] = page
+
+        endpoint = 'pages/{}/incidents'.format(self._page_id)  # pragma: allowlist secret
+        endpoint += '?{}'.format(parameters)
+
+        ret_val, response = self._make_rest_call(endpoint, action_result, headers=headers, params=parameters)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        for incident in response:
+            action_result.add_data(incident)
+
+        summary = action_result.set_summary({})
+        summary['num_incidents'] = len(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_create_incident(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        api_key = "API Key {}".format(self._api_key)  # pragma: allowlist secret
+        headers = {'Content-Type': 'application/json', 'Authorization': api_key}
+
+        data = dict()
+
+        components = dict()
+        ret_val, components = self._get_fields(param, action_result)
+
+        data["incident"] = {
+            "name": param.get('name'),
+            "status": param.get('status'),
+            "impact_override": param.get('impact_override'),
+            "body": param.get('body'),
+            "component_ids": param.get('component_ids'),
+            "components": components
+        }
+
+        endpoint = 'pages/{}/incidents'.format(self._page_id)  # pragma: allowlist secret
+        ret_val, response = self._make_rest_call(endpoint, action_result, json=data, headers=headers, method="post")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully created incident")
+
+    def _handle_update_incident(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        api_key = "API Key {}".format(self._api_key)  # pragma: allowlist secret
+        headers = {'Content-Type': 'application/json', 'Authorization': api_key}
+
```

Comment:
can we move this block to `_make_rest_call` if possible?

Reply:
Done